### PR TITLE
Bump Safari iOS 15 engine version

### PR DESCRIPTION
**Pull request by queengooborg**
This PR bumps the engine version for Safari iOS 15 to match Safari Desktop 15.

### CODE REVIEW

**Comment by foolip** [comment]
> a solution I know we have done before.

This made me curious, so I looked up which PRs have previously modified the engine versions and found https://github.com/mdn/browser-compat-data/pull/4470 + https://github.com/mdn/browser-compat-data/pull/4732 + https://github.com/mdn/browser-compat-data/pull/11146.

But none of these say anything about mirroring. Do you still remember what the change was, so we can perhaps revert it after we fix the mirroring in another way?

**Comment by queengooborg** [comment]
Oh huh, I could've sworn one of those involved a tweak to fix up mirroring...  I must have been thinking of #11333, but that was removing excessive parts of the version number.

This is probably a good incentive to implement the `upstream_version` property I mentioned earlier, before adopting build-time mirroring more.

File: browsers/safari_ios.json [hunk 1]
```diff
@@ -149,7 +149,7 @@
           "release_notes": "https://developer.apple.com/documentation/safari-release-notes/safari-15-release-notes",
           "status": "retired",
           "engine": "WebKit",
-          "engine_version": "612.1.27"
+          "engine_version": "612.1.29"
```
Comment (gsnedders) [inline browsers/safari_ios.json:152]: https://github.com/WebKit/WebKit/blob/releases/Apple/Safari-15-iOS-15.0/Source/WebKit/Configurations/Version.xcconfig claims this change is wrong.

File: scripts/release/mirror.test.js [hunk 2]
```diff
@@ -76,7 +76,7 @@ describe('mirror', () => {
         ['13.1', '13.4'],
         ['14', '14'],
         ['14.1', '14.5'],
-        ['15', '15.1'],
+        ['15', '15'],
```
Comment (gsnedders) [inline scripts/release/mirror.test.js:79]: …but this change seems correct.

File: browsers/safari_ios.json (hunk 1 above)
Comment (foolip) [reply browsers/safari_ios.json:152]: Thanks for confirming!

File: scripts/release/mirror.test.js (hunk 2 above)
Comment (foolip) [reply scripts/release/mirror.test.js:79]: We'll have to fix it in some other way.

Comment (foolip) [reply scripts/release/mirror.test.js:79]: https://github.com/mdn/browser-compat-data/pull/16594

**Pull request closed without merging**